app/controllers/purchases_controller.rb

```diff
@@ -10,15 +10,14 @@ def new
 
   def create
     authorize Purchase
-    iugu = Iugu::Integration.new(token: ENV['IUGU_API_TOKEN'])
     @course = Course.find(params[:purchase][:course])
     return render_error('o curso escolhido') unless @course
+    iugu = Iugu::Integration.new(token: ENV['IUGU_API_TOKEN'])
     iugu_charge = create_iugu_charge(iugu, @course)
-    return render_error('sua cobrança') unless iugu_charge
-    purchase = create_purchase(
+    return render_error('sua cobrança') unless iugu_charge&.[]('success')
+    create_purchase_and_redirect(
```

Comment:
I had to put the redirect inside the method because Rubocop was complaining about `Metrics/ABCSize` and I didn't find any better ways to improve it.

Reply:
It became harder to understand. I suggest you create the following `protected` method on `ApplicationController`:

```
def iugu
  @iugu ||= Iugu::Integration.new(token: ENV['IUGU_API_TOKEN'])
end
```

Reply:
Then you can avoid passing the `iugu` parameter to `create_iugu_charge` and other similar methods. I think this will help with ABC size.